### ISSUE TITLE
Missing session handler when using token-store:cookie configuration

### ISSUE THREAD

**Issue by RasmusOlesen** [ahus1/keycloak-dropwizard-integration#38]
In de.ahus1.keycloak.dropwizard.KeycloakBundle#run, line52, the comment states that no session is needed when using the "token-store:cookie" configuration.

But later in de.ahus1.keycloak.dropwizard.KeycloakAuthFilter#validateRequest, line 121, the JSESSIONID cookie is always set, unless the client is configured to "BearerOnly".

Resulting in the following error:

```
ERROR [2020-08-24 12:51:25,122] io.dropwizard.jersey.errors.IllegalStateExceptionMapper: Error handling a request: ba8a53f8e9f4ca07
! java.lang.IllegalStateException: No SessionManager
! at org.eclipse.jetty.server.Request.getSession(Request.java:1605)
! at org.eclipse.jetty.server.Request.getSession(Request.java:1581)
! at de.ahus1.keycloak.dropwizard.KeycloakAuthFilter.validateRequest(KeycloakAuthFilter.java:121)
! at de.ahus1.keycloak.dropwizard.KeycloakAuthFilter.filter(KeycloakAuthFilter.java:54)
! at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:108)
! at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:44)
! at org.glassfish.jersey.process.internal.Stages.process(Stages.java:173)
! at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:245)
! at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
! at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
! at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
! at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
! at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
! at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
! at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:232)
! at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680)
! at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:394)
! at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:346)
! at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:366)
! at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:319)
! at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205)
! at io.dropwizard.jetty.NonblockingServletHolder.handle(NonblockingServletHolder.java:50)
! at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1631)
! at io.dropwizard.servlets.ThreadNameFilter.doFilter(ThreadNameFilter.java:35)
! at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
! at io.dropwizard.jersey.filter.AllowedMethodsFilter.handle(AllowedMethodsFilter.java:47)
! at io.dropwizard.jersey.filter.AllowedMethodsFilter.doFilter(AllowedMethodsFilter.java:41)
! at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618)
! at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:549)
! at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
! at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:602)
! at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
! at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
! at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1369)
! at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
! at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:489)
! at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
! at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1284)
! at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
! at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
! at com.codahale.metrics.jetty9.InstrumentedHandler.handle(InstrumentedHandler.java:249)
! at io.dropwizard.jetty.ContextRoutingHandler.handle(ContextRoutingHandler.java:37)
! at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:767)
! at org.eclipse.jetty.server.handler.RequestLogHandler.handle(RequestLogHandler.java:54)
! at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:173)
! at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
! at org.eclipse.jetty.server.Server.handle(Server.java:501)
! at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
! at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:556)
! at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
! at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:272)
! at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
! at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
! at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
! at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
! at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
! at java.lang.Thread.run(Thread.java:748)
```

**Comment by ahus1** [comment]
Hi @RasmusOlesen, thanks for looking into this and reporting it. 

There seems to be a problem in the code and a missing unit test for this. 

Do you want to give it a try and provide a test case and updated code in a pull request? 

You could also prepare a pull request and test manually first. I could then later see how a test could be automated for this.

Please let me know how you want to proceed.

**Comment by RasmusOlesen** [comment]
Hi @ahus1 

I might be able to find some time during the next couple of weeks, so i will give it a shot.

/Rasmus

**Comment by RasmusOlesen** [comment]
Hi @ahus1 
I haven't managed to get a local build environment to work, so i can't test my "solution"

But i am pretty sure that the following change to de.ahus1.keycloak.dropwizard.KeycloakAuthFilter#validateRequest, should solve the problem. (line 119)
`            
if (!adapterConfig.isBearerOnly() && deployment.getTokenStore() == TokenStore.SESSION) {
`

**Comment by ahus1** [comment]
I just published version 1.2.0 including this change, should be available on Maven central soon.
